woocommerce/woocommerce-ios
# [Woo POS] Extract totals logic to `TotalsViewModel`

Part of https://github.com/woocommerce/woocommerce-ios/issues/12998


## Description

This PR is part 3 (and last) of splitting `PointOfSaleDashboardViewModel` responsibilities based on the 3 main views in POS: 
* Item selector 
* Cart view
* Totals view (this PR)

As order syncing and payment collection work hand in hand, the move of both parts is in the same PR to also unblock other work affected by the view models like Bozidar's UI updates https://github.com/woocommerce/woocommerce-ios/pull/13166. 

Most of the diffs move code from `PointOfSaleDashboardViewModel` to `TotalsViewModel` with minimum changes and reorganization in the file to privatize the methods as much as possible. 

`TotalsView` still depends on `PointOfSaleDashboardViewModel` for starting a new transaction and to provide the cart items & POS items for calculating the total amount manually after an order sync failure, we can consider refactoring in the future.

## Changes
- Created `TotalsViewModel`, which encapsulates most of totals and payments logic
- CurrencyFormatter is injected from the app to the pos through the entry point
- Cleanup of unused code
- Added empty preliminary list of unit test, that will be tackled on a separate PR

## Testing


Prerequisite: the store is eligible for POS and the feature switch is enabled in Menu > Settings > Experimental Features > POS
* Launch app
* Go to Menu > Point of Sale
* Add an item (with a non-zero price and ideally with tax applied on checkout) to cart
* Tap `Checkout` --> the order total should be updated to a correct, non-zero value
* Tap `Collect Payment` and proceed to tap a card to pay --> the transaction should succeed
* Tap `New transaction` and repeat the above steps --> the second transaction should succeed
* Tap `Exit POS` and go to the orders tab --> the two orders should be shown as completed
